mmap/src/lib.rs

```diff
@@ -155,6 +155,62 @@ enum Address {
     Onto(usize),
 }
 
+/// The error condition
+///
+/// This type is mostly a wrapper for `std::io::Error` with one additional
+/// feature: it conveys ownership to a mapping. This enables the pattern
+/// where an old mapping is valid until the conversion operation is successful.
+/// If the operation is unsuccessful, the old mapping is returned along with
+/// the error condition.
+#[derive(Debug)]
+pub struct Error<T> {
+    /// The previous mapping that could not be modified
+    pub map: T,
+
+    /// The underlying error
+    pub err: std::io::Error,
+}
+
+impl<T> std::fmt::Display for Error<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        self.err.fmt(f)
+    }
+}
+
+impl<T: std::fmt::Debug> std::error::Error for Error<T> {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        Some(&self.err)
+    }
+
+    fn cause(&self) -> Option<&dyn std::error::Error> {
```

Comment:
you don't have to implement this... it's already implemented in the trait using `source`

Reply:
fixed